dotcom-rendering/src/web/components/KeyEventsContainer.tsx

```diff
@@ -0,0 +1,31 @@
+import KeyEvents, {
+	KeyEvent,
+} from '@guardian/common-rendering/src/components/keyEvents';
+
+type Props = {
+	keyEvents: Block[];
+	format: ArticleFormat;
+};
+
+export const KeyEventsContainer = ({ keyEvents, format }: Props) => {
+	const transformedKeyEvents: KeyEvent[] = keyEvents.reduce(
+		(acc: KeyEvent[], keyEvent) => {
+			if (keyEvent.title && keyEvent.blockFirstPublishedDisplay) {
+				acc.push({
```

Comment:
interested why you favoured a reduce over a filter then a map?

Reply:
Great shout! Refactored 👍